### PR TITLE
TEN-82 added Enrol-current-user view

### PR DESCRIPTION
**Pull request by Chrissystk**
To test the page locally use the gg-stub (http://localhost:9949/auth-login-stub/gg-sign-in) and in the redirectUrl put "http://localhost:7750/tax-enrolment-assignment-frontend/enrol-pt/enrol-current-user-id". This will then present the page for a fixed user that has SA on another account

### CODE REVIEW

File: it/controllers/EnrolCurrentUserControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+          s"return $UNAUTHORIZED" in {
+            stubAuthorizePostUnauthorised(failureReason)
+            stubPost(s"/write/.*", OK, """{"x":2}""")
+
+            val res = buildRequest(urlPath)
+              .withHttpHeaders(xSessionId, csrfContent)
+              .get()
+
+            whenReady(res) { resp =>
+              resp.status shouldBe UNAUTHORIZED
```
Comment (Ufuoma-Daniel) [inline it/controllers/EnrolCurrentUserControllerISpec.scala:92]: Unsure about having two tests if they're the exact same thing. If the point is to cover the different Exception types of NoActiveSession and AuthorisationException then this becomes an incorrect test. SessionRecordNotFound should be a 303 status to the login page right?

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/EnrolCurrentUser.scala.html [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    @govukRadios(Radios(
+        fieldset = Some(Fieldset(
+          legend = Some(Legend(
+          content = Text(messages("enrolCurrentUser.heading")),
+          classes = "govuk-fieldset__legend--l",
+          isPageHeading = true
+          ))
+        )),
+        errorMessage = formError.map(error => ErrorMessage(
+          content = Text(messages("ers_check_file_type.alert"))
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/EnrolCurrentUser.scala.html:59]: Assuming this error key is meant to be something else?

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/EnrolCurrentUser.scala.html [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    pageTitle = Some(messages("enrolCurrentUser.title")), Some("?")) {
+
+    @if(formError.isDefined) {
+        @govukErrorSummary(ErrorSummary(
+          classes = "validation-summary govuk-!-display-none",
+          errorList = Seq(
+          ErrorLink(
+            href = Some("#use-current-id-error"),
+            content = Text(messages(formError.get))
+          )
```
[the remark this answers is not in the document]
Comment (Chrissystk) [reply app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/EnrolCurrentUser.scala.html:43]: Yes this page can only ever have one error

**Comment by Ufuoma-Daniel** [comment]
Ooh minor aside, can you up our coverage threshold with this ticket since it's still at 77% but we are actually at 85% coverage. Also, up the minor dependencies:
```
uk.gov.hmrc:http-caching-client              : 9.5.0-play-28           -> 9.6.0-play-28
uk.gov.hmrc:play-frontend-hmrc             : 3.4.0-play-28           -> 3.5.0-play-28
```

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/EnrolCurrentUser.scala.html (hunk 2 above)
Comment (Chrissystk) [reply app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/EnrolCurrentUser.scala.html:59]: yep

File: it/controllers/EnrolCurrentUserControllerISpec.scala (hunk 1 above)
Comment (Chrissystk) [reply it/controllers/EnrolCurrentUserControllerISpec.scala:92]: These are not the NoActiveSession. One is no session being present full stop and the other is AuthorisationExceptions for confidence level (we have CL200 rule). The authCode for AuthorisationExceptions returns an unauthorised in both these cases. Also you have this same function for testing is in the LandingPage tests.

Comment (Ufuoma-Daniel) [reply it/controllers/EnrolCurrentUserControllerISpec.scala:92]: Is the replicable behaviour? I thought no session being present always returned a NoActiveSession, I haven't ever seen it return an AuthorisationException. Happy to leave this as is, but regardless, but it'd be good to confirm.

Comment (Chrissystk) [reply it/controllers/EnrolCurrentUserControllerISpec.scala:92]: The error returned is SessionRecordNotFound

**Comment by Chrissystk** [comment]
updated

**Review by Ufuoma-Daniel — APPROVED** [review]

**Pull request merged**